src/ripple/app/consensus/LedgerConsensus.cpp

```diff
@@ -372,7 +374,8 @@ class LedgerConsensusImp
       @param map      the transaction set.
       @param acquired true if we have acquired the transaction set.
     */
-    void mapComplete (uint256 const& hash, SHAMap::ref map, bool acquired)
+    void mapComplete (uint256 const& hash, std::shared_ptr<SHAMap> const& map,
+                      bool acquired)
```

Comment:
Hell no. We really need to end the practice of these "convenience" nested types. They do nothing but create type proliferation and make it impossible to forward declare key classes. They confuse readers who have to look up the type. `std::shared_ptr<SHAMap> const&` is much more concise than `SHAMap::ref`, `SHAMap::sref`, or any other nested type alias.

The definition of verbosity is key here. I define verbose as using three types (`SHAMap`, `SHAMap::ref`, `std::shared_ptr`) where two would suffice (`SHAMap`, `std::shared_ptr`). No one will ever be confused by seeing `std::shared_ptr<SHAMap> const&`. The meaning is unambiguous. And we can use `std::shared_ptr<SHAMap>` with only a forward declaration of `SHAMap`.